app/controllers/animals_controller.rb

```diff
@@ -1,15 +1,18 @@
 class AnimalsController < ApplicationController
   def new
     @animal = Animal.new
+    @animal_types = AnimalType.all
   end
 
   def create
     @animal = Animal.new(animal_params)
+    @animal.user_id = current_user.id
```

Comment:
```suggestion
   current_user.animals.new(animal_params)
```

Daria pra trocar essa linha e a de cima